db-shared/src/main/java/com/splicemachine/db/shared/ProgressInfo.java

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright (c) 2020 - 2021 Splice Machine, Inc.
+ *
+ * This file is part of Splice Machine.
+ * Splice Machine is free software: you can redistribute it and/or modify it under the terms of the
+ * GNU Affero General Public License as published by the Free Software Foundation, either
+ * version 3, or (at your option) any later version.
+ * Splice Machine is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU Affero General Public License for more details.
+ * You should have received a copy of the GNU Affero General Public License along with Splice Machine.
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package com.splicemachine.db.shared;
+
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+
+/**
+ *  Information about the progress of a running operation.
+ *  JOB: exactly ONE job running at each time
+ *  JOB is sequence of STAGES:
+ *       STAGE exactly ONE stage running at each time
+ *       STAGE is a sequence of TASKS:
```

Comment:
There could be more than one job and/or stage running at the same time.
When we have multiple (native spark) broadcast joins we should have multiple concurrent jobs at once, one for each broadcast resultset.
For stages, some Spark operations introduce concurrency, for instance both branches of a mergeSortJoin can run concurrently

Reply:
Just tested it with a mergeSortJoin and it appends both progress bars to the same line, resulting in something like this:
```
splice> select count(*) from a natural join a;
[Running ...                                                                                        ]
Produce Result Set: select count(*) from a natural join a (Job 2, Stage 1 of 4)
[-----------------+----------25%---+----------25%-----------+----------25%----------+-----------50%--------25%----------+-----------50%-----------+-----------+-----------50%-----------+-------------+-----------50%-----------+----------75%---+-----------50%-----------+----------75%---------------50%-----------+----------75%-----------+-0%-----------+----------75%-----------+------------+----------75%-----------+------
```

Reply:
With 2 jobs it fails with this exception:
```
create table a(i int); -- insert 12582912 rows into a
select * from a inner join (select count(*) as i from a
> ) b --splice-properties joinStrategy=broadcast
> on a.i = b.i
> inner join (select count(*) as i from a
> ) c --splice-properties joinStrategy=broadcast
> on a.i = c.i;
[Running ...                                                                                        ]

Exception while parsing/updating ProgressInfo: java.lang.NumberFormatException: For input string: ")"
```